presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -154,6 +156,13 @@ else if (config.getPinGlueClientToCurrentRegion()) {
             }
         }
 
+        if (config.getIamRole().isPresent()) {
+            AWSCredentialsProvider credentialsProvider = new STSAssumeRoleSessionCredentialsProvider
+                    .Builder(config.getIamRole().get(), "presto-session")
```

Comment:
Should we give a separate `roleSessionName` here to mark this session as used for glue access?